### ISSUE TITLE
Can't progress in Deep Jungle

### ISSUE THREAD

**Issue by cmanr010** [Denhonator/KHPCSpeedrunTools#88]
I have managed to find all 6 slides throughout the worlds. I started progressing in Deep Jungle before progressing in Hollow Bastion and I got up to the point where Jane asks you to find the slides. I remember Clayton was standing outside of the tent and no heartless spawned.

Eventually I progressed through Hollow Bastion fully and found the last slide I needed there. I tried to go back to Deep Jungle to finish it, but Jane just keeps telling me there are 6 slides left to find. I can't interact with the projector, the interaction prompt doesn't come up. Also, Clayton is no longer outside and heartless spawn everywhere just like it normally would when you finish Hollow Bastion in the original game (which implies Deep Jungle was finished). I tried to go to the Treehouse and nothing happened. I just can't seem to interact with the projector with the slides to progress Deep Jungle, so I'm left with it incomplete at the moment.

Because of this, I also am unable to finish Traverse Town since I can't interact with Red Trinity marks.

**Comment by Denhonator** [comment]
Completing HB does mess with the progression of other worlds, unfortunately, so in absence of a better system, players usually avoid beating HB too early, but if the last slide required beating HB, then it's clearly a logic error. Can you tell me where it was?

**Comment by cmanr010** [comment]
Honestly I don't think it required beating HB. It was in one of the Lift Stop treasure chests though and either I opened it before completing finishing HB and I didn't happen to return to Deep Jungle to finish it then, or I opened it sometime after finishing HB and I happened to return to Deep Jungle to try to finish it then and that's when I noticed I could no longer progress there. Is there a way I can get a log to show you exactly where it was? Also is there a way to reset the Deep Jungle "flags" to make it so that I can continue it or no?

**Comment by Denhonator** [comment]
There should be a spoilers.txt in the game folder (or LuaFrontend folder if you use that) or you could send the seed itself.

Unfortunately I never developed a system to fix those flags. I could probably manage to do it manually, but that's a lot of effort just to fix one seed

**Comment by cmanr010** [comment]
[seed.txt](https://github.com/user-attachments/files/16553011/seed.txt)
[spoilers.txt](https://github.com/user-attachments/files/16553012/spoilers.txt)

Ok I've attached the seed and spoilers file.

**Comment by gaithern** [comment]
I developed a script to combat this issue for KH1 AP Rando, it should work here?
https://github.com/gaithern/KH-1FM-AP-LUA/blob/main/1fmAPFixRiku2.lua

**Comment by cmanr010** [comment]
> I developed a script to combat this issue for KH1 AP Rando, it should work here? https://github.com/gaithern/KH-1FM-AP-LUA/blob/main/1fmAPFixRiku2.lua

Cool, what exactly does it do? Do I just put it in the script folder? Also does this only work for new saves or could it help fix the issue in my current save?

**Comment by gaithern** [comment]
Basically it checks to see if you have completed the Riku-Ansem fight and if one of the following worlds are in an incomplete state:

- Wonderland
- Deep Jungle
- Agrabah
- Neverland

If they are, it reverts those world's room flags back to the state they should be in based on their story progression byte (which is unaffected by that fight).  After you complete the world, it then sets the rooms back to their post Riku-Ansem state.

You can just put it in the scripts folder

Should work for your current save.

**Comment by cmanr010** [comment]
> Basically it checks to see if you have completed the Riku-Ansem fight and if one of the following worlds are in an incomplete state:
> 
> * Wonderland
> * Deep Jungle
> * Agrabah
> * Neverland
> 
> If they are, it reverts those world's room flags back to the state they should be in based on their story progression byte (which is unaffected by that fight). After you complete the world, it then sets the rooms back to their post Riku-Ansem state.
> 
> You can just put it in the scripts folder
> 
> Should work for your current save.

Thanks for the script! I see some progress now. I will list the observations and steps I took to get to the cutscene after running the 6 slides on the projector.

Now, Clayton is standing outside in the camp area again, no heartless are spawning, and Sabor spawns as normal. However, I still can't interact with the projector. I talk to Jane and she says there are still 6 slides to collect. I go to the Treehouse and wait until Sabor kills me to see if that does anything and it does not. Then I go back to the tent and walk outside and notice that the slides are physically there around the camp when they weren't previously. I collect them (even though I already had 1 of each in my inventory) and then am able to interact with the projector to proceed.

**Comment by gaithern** [comment]
Yeah there is some flag somewhere to track how many slides you have picked up, I think the game looks for that rather than how many you have in your inventory, but don't recall exactly.  Glad its working now!

**Comment by Denhonator** [comment]
Iirc the way I implemented it, if the slides are randomized, is that you need to find them and after that you can collect them at the camp because of those flags

**Comment by cmanr010** [comment]
> Yeah there is some flag somewhere to track how many slides you have picked up, I think the game looks for that rather than how many you have in your inventory, but don't recall exactly. Glad its working now!

Thanks again! On a side note, I noticed your script has a line about Kurt Zisa. I had actually started a second save file yesterday because I thought my original one was a lost cause and I got to Olympus Coliseum and lo and behold, my Preliminaries Cloud fight was replaced with Kurt Zisa! Obviously I couldn't kill him. Does your script also help make it so Kurt Zisa doesn't replace other bosses?

**Comment by gaithern** [comment]
No, the line about Kurt Zisa is just making sure that the carpet appears to take you to him when you complete Agrabah if you've beaten Riku-Ansem, specifically if you beat Agrabah after Riku-Ansem.

**Comment by cmanr010** [comment]
> No, the line about Kurt Zisa is just making sure that the carpet appears to take you to him when you complete Agrabah if you've beaten Riku-Ansem, specifically if you beat Agrabah after Riku-Ansem.

Ok thank you. Reporting back with a follow up issue. I did everything in Deep Jungle, climbed up the waterfall cavern, and the cutscene played. It took me back to the tent afterwards and I boarded the gummi ship and saw the dialogue play where they decide to go to back to Traverse Town. However, on the map screen it doesn't show that I unlocked the Deep Jungle keyhole. Also I did not get the ability to interact with Red Trinities.

**Comment by cmanr010** [comment]
This is the last journal entry I see for Deep Jungle at the moment.

![deepjungle](https://github.com/user-attachments/assets/035e75a2-e937-433e-ba5f-498774769e40)

**Comment by gaithern** [comment]
That I'm not sure about, I believe trinities are randomized so did Red Trinity end up being vanilla?

**Comment by cahudson94** [comment]
Yeah trinities should be random so you may have unlocked another the keyhole not showing as completed and the journal being in that state seems wonky.

**Comment by cmanr010** [comment]
> Yeah trinities should be random so you may have unlocked another the keyhole not showing as completed and the journal being in that state seems wonky.

Yesterday I started a new save file and got through Deep Jungle. I noticed that after going up the waterfall cavern and procing the cutscene up there, there was another cutscene that played where Tarzan, Jane, and Sora's gang are in the tent talking and Tarzan gives Sora what I assume is originally supposed to be the keychain of that world. Then he gave me the ability to do Red Trinities. However in my original randomized playthrough this cutscene didn't happen. I went up the cavern, had the cutscene there, then the game teleported me to the tent but that cutscene I just mentioned didn't play out, so I didn't get whatever that randomized item would have been from the keychain nor did I get the ability to do Red Trinities. It seems there is something bugged with my original file.

**Comment by cahudson94** [comment]
yeah I would guess there is another flag that was not quite caught by that second mod script.
